zetaclient/chains/evm/observer/v2_inbound_tracker.go

```diff
@@ -2,19 +2,24 @@ package observer
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
+	ethcommon "github.com/ethereum/go-ethereum/common"
 	ethtypes "github.com/ethereum/go-ethereum/core/types"
 	"github.com/onrik/ethrpc"
 	"github.com/zeta-chain/protocol-contracts/v2/pkg/gatewayevm.sol"
 
 	"github.com/zeta-chain/node/zetaclient/zetacore"
 )
 
+var errEventNotFound = errors.New("no gateway event found in inbound tracker")
+
 // ProcessInboundTrackerV2 processes inbound tracker events from the gateway
 func (ob *Observer) ProcessInboundTrackerV2(
 	ctx context.Context,
 	gateway *gatewayevm.GatewayEVM,
+	gatewayAddr ethcommon.Address,
```

Comment:
Why do we need to pass `gateway` and `gatewayAddr` into this function? It should be within `*Observer`